### PR TITLE
feat: migrate to Vite

### LINKED ISSUE #31: Research how to reduce the size of the container image
Check if 'Vite' can build and bundle the code for the console, serving it using Nginx, and reduce the image size

### PR DESCRIPTION
**Pull request by vbelouso**
- Move from `Create React App` to `Vite` for faster builds, improved performance, and modern tooling support.
- Add application bundle that is to be served via `Nginx`
- Add Nginx configuration template for the application serving
- Remove unused packages, scripts, assets
- Update Containerfile to enhance reliability and align with best practices.

Closes #31

### CODE REVIEW

**Comment by r2dedios** [comment]
Closes #31

File: Containerfile [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
-####################
-# Obtained from: https://catalog.redhat.com/software/containers/ubi9/nodejs-18/62e8e7ed22d1d3c2dfe2ca01?container-tabs=gti
-FROM registry.access.redhat.com/ubi9/nodejs-18:9.5-1733148761
+ARG NODE_VERSION=18
+ARG NODE_IMAGE_TAG=9.5-1733148761
 
-# Labels
-LABEL VERSION="v0.2"
-LABEL description="ClusterIQ Web Console"
+ARG NGINX_VERSION=124
```
Comment (r2dedios) [inline Containerfile:4]: Quay.io doesn't support ARGS, but as long as we don't need to setup any specific value during the building, it should work with the default values.

File: Makefile [hunk 2]
```diff
@@ -44,14 +42,12 @@ build-local:
 
 push:
 	@$(CONTAINER_ENGINE) push $(CONSOLE_IMAGE):latest
-	@$(CONTAINER_ENGINE) push $(CONSOLE_IMAGE):${VERSION}
 	@$(CONTAINER_ENGINE) push $(CONSOLE_IMAGE):${IMAGE_TAG}
 
-
-start-dev: export REACT_APP_CIQ_API_URL = http://localhost:8081
+start-dev: export VITE_CIQ_API_URL = http://localhost:8081
```
Comment (r2dedios) [inline Makefile:47]: Here we are defining the Backend URL var name as "VITE_CIQ_API_URL"
However, the var used on the `nginx.conf` file is called: "BACKEND_URL"

File: Makefile [hunk 3]
```diff
@@ -44,14 +42,12 @@ build-local:
...
 start-dev:
 	@echo "### [Starting project DEV MODE] ###"
-	@echo "### [API-URL: $$REACT_APP_CIQ_API_URL] ###"
+	@echo "### [API-URL: $$VITE_CIQ_API_URL] ###"
 	@npm run start
```
Comment (r2dedios) [inline Makefile:51]: This is correct for running the console on local.
If we want to adapt this to run as a container (like the backend) we should replace this by a podman/docker command. 

As the backend is deployed using compose files, we need to deploy the console container attached to the compose network


I tried it by my own, and including these changes should fix this:
```
start-dev: export BACKEND_URL = http://api:8080
start-dev:
  @echo "### [Starting project DEV MODE] ###"
  @echo "### [API-URL: $$BACKEND_URL] ###"
  @podman run -dt --name console --network docker-compose_default -e BACKEND_URL=$$BACKEND_URL -p 8080:8080 quay.io/ecosystem-appeng/cluster-iq-console:latest
```

**Review by r2dedios — CHANGES_REQUESTED** [review]

File: Makefile [hunk 4]
```diff
@@ -44,14 +42,12 @@ build-local:
...
```
Comment (r2dedios) [inline Makefile:52]: Missing "stop-dev" command
```
stop-dev:
  podman rm -f console

```

Comment (r2dedios) [reply Makefile:52]: npm ERR! Missing script: "lint"
npm ERR! Missing script: "format"

When running: "make lint format"

Do we still using these commands?

**Review by r2dedios — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -2,54 +2,28 @@
...
-    "react-scripts": "5.0.1",
-    "typescript": "^4.9.5",
-    "web-vitals": "^2.1.4"
-  },
+  "type": "module",
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject"
```
Comment (r2dedios) [inline package.json:35]: Missing "lint" and "format" scripts. Not sure if we still needing them. If not, it's ok, but we should remove the Makefile rules for those verbs too.

**Review by r2dedios — CHANGES_REQUESTED** [review]

**Comment by r2dedios** [comment]
```
diff --git a/Makefile b/Makefile
index 68f18bb..b890608 100644
--- a/Makefile
+++ b/Makefile
@@ -14,8 +14,8 @@ Makefile Rules:
 	compile: Builds the Console on local
 	build: Builds the Console on a Container image
 	push: Pushes every container image into remote repo
-	start-dev: Starts a local environment using 'docker/$(CONTAINER_ENGINE)-compose'
-	test: Runs test
+	start-dev: Starts a local environment using '$(CONTAINER_ENGINE)-compose'
+	test: Runs tests
 	checks: Runs linters and format
 	help: Displays this message
 endef
@@ -44,11 +44,14 @@ push:
 	@$(CONTAINER_ENGINE) push $(CONSOLE_IMAGE):latest
 	@$(CONTAINER_ENGINE) push $(CONSOLE_IMAGE):${IMAGE_TAG}
 
-start-dev: export VITE_CIQ_API_URL = http://localhost:8081
+start-dev: export BACKEND_URL = http://api:8080
 start-dev:
 	@echo "### [Starting project DEV MODE] ###"
-	@echo "### [API-URL: $$VITE_CIQ_API_URL] ###"
-	@npm run start
+	@echo "### [API-URL: $$BACKEND_URL] ###"
+	@podman run -dt --name console --network docker-compose_default -e BACKEND_URL=$$BACKEND_URL -p 8080:8080 quay.io/ecosystem-appeng/cluster-iq-console:latest
+
+stop-dev:
+	@podman rm -f console
 
 test: checks
 	@echo "### [Running tests] ###"
@@ -64,4 +67,4 @@ format:
 
 .DEFAULT_GOAL := help
 help:
-	echo "$$HELP_MSG"
+	@echo "$$HELP_MSG"
```

File: Makefile (hunk 2 above)
Comment (vbelouso) [reply Makefile:47]: I did this on purpose to separate the prod and dev environments + not expose the variable to the client

https://vite.dev/guide/env-and-mode#env-files

File: Makefile (hunk 3 above)
Comment (vbelouso) [reply Makefile:51]: It seems that a separate Makefile target is necessary to launch the frontend specifically in the container. 
I would reserve `start-dev` exclusively for launching without a container.

File: package.json (hunk 5 above)
Comment (vbelouso) [reply package.json:35]: Makefile updated with some additional improvements, PTAL

File: Makefile (hunk 4 above)
Comment (vbelouso) [reply Makefile:33]: Answered at https://github.com/RHEcosystemAppEng/cluster-iq-console/pull/40#discussion_r1896079887

**Review by r2dedios — APPROVED** [review]
LGTM

**Pull request merged**